File: android/app/src/main/java/com/rafaeltmbr/stopwatch/domain/data/repositories/impl/StopwatchRepositoryImpl.kt
package com.rafaeltmbr.stopwatch.domain.data.repositories.impl

import com.rafaeltmbr.stopwatch.domain.data.data_sources.StopwatchDataSource
import com.rafaeltmbr.stopwatch.domain.data.repositories.StopwatchRepository
import com.rafaeltmbr.stopwatch.domain.entities.StopwatchState


class StopwatchRepositoryImpl(
    private val dataSource: StopwatchDataSource
) : StopwatchRepository {
    override suspend fun load() = dataSource.load()

    override suspend fun save(state: StopwatchState) = dataSource.save(state)
}

File: ios/Stopwatch/domain/data/repositories/impl/StopwatchRepositoryImpl.swift
struct StopwatchRepositoryImpl: StopwatchRepository {
    private let stopwatchDataSource: StopwatchDataSource
    
    init(_ stopwatchDataSource: StopwatchDataSource) {
        self.stopwatchDataSource = stopwatchDataSource
    }
    
    func load() async -> StopwatchState? {
        return await stopwatchDataSource.load()
    }
    
    func save(_ state: StopwatchState) async {
        await stopwatchDataSource.save(state)
    }
}
